fsharp-backend/src/Wasm/Wasm.fs

```diff
@@ -320,22 +320,39 @@ type EvalWorker =
           )
           Error($"exception: {e.Message}, metdata: {metadata}")
 
-      match args with
-      | Error e -> return Error e
-      | Ok args ->
+      let! result =
+        task {
+          match args with
+          | Error e -> return Error e
+          | Ok args ->
+            try
+              let! result = Eval.performAnalysis args
+              return Ok result
+            with
+            | e ->
+              let metadata = Exception.nestedMetadata e
+              System.Console.WriteLine("Error running analysis in Blazor")
+              System.Console.WriteLine($"called with message: {message}")
+              System.Console.WriteLine(
+                $"caught exception: \"{e.Message}\" \"{metadata}\""
+              )
+              return Error($"exception: {e.Message}, metadata: {metadata}")
+        }
+
+      let serialized =
+        let serialize = Json.Vanilla.serialize
+
         try
-          let! result = Eval.performAnalysis args
-          return Ok result
+          serialize result
         with
         | e ->
           let metadata = Exception.nestedMetadata e
-          System.Console.WriteLine("Error running analysis in Blazor")
+          System.Console.WriteLine("Error serializing results of Blazor analysis")
           System.Console.WriteLine($"called with message: {message}")
           System.Console.WriteLine(
             $"caught exception: \"{e.Message}\" \"{metadata}\""
           )
-          return Error($"exception: {e.Message}, metadata: {metadata}")
+          serialize (Error($"exception: {e.Message}, metadata: {metadata}"))
```

Comment:
this is the other call to `serialize` I'm referring to.

Reply:
Ah, I missed this. Might be clearer by making both explicit, but could go either way.